### ISSUE TITLE
Appendix C- of symmetric sub-spec uses AES-CBC as example with unavailable capabilities

### ISSUE THREAD

**Issue by hbooth** [usnistgov/ACVP#121]
If you look at the examples in Appendix C of the symmetric sub-spec, one them uses AES-CBC as an example for an MCT test result. I am guessing a copy/paste error has resulted in some extra capabilities which are not valid for the AES-CBC algorithm.

**Comment by atvassilev** [comment]
Reclassifying as a bug.

**Comment by bfussell** [comment]
Not sure I follow this one, can you be a bit more specific.

**Comment by hbooth** [comment]
The second and third examples (extracted below) in Appendix C both mention ivLen and iv as well as ptLen. CAVs currently does not have a ptLen for AES-CBC, although I think we may want to consider changing it for ACVP. AES-CBC does not appear to have an IV parameter in CAVs either.

```json
              [
                { "acvVersion": "0.3" },
                { "vsId": 1564,
                  "algorithm": "AES-CBC",
                  "testGroups": [
                    {
                      "direction": "encrypt",
                      "keyLen": 128,
                      "ivLen": 128
                      "ptLen": 128
                      "testType" : "MCT",
                      "tests": [
                        {
                          "tcId": 3171,
                          "key": "868a6705c03e630327259cd09e63354a",
			  "iv" : "a771c0fc42300fa7b4c1eca82e1eb2f9",
                          "pt": "8466e3b1680b40f8bae488e2f392ab33"
                        }
                      ]
                    }
		  ]
	        }
              ]
```

```json
              [
                  { "acvVersion": "0.3" },
                  { "vsId": 1564,
                    "testResults": [
                        {
			  "tcId": 3171,
			  "resultsArray": [
			    {
			      "count" : 0,
			      "key" : "868a6705c03e630327259cd09e63354a",
			      "iv": "a771c0fc42300fa7b4c1eca82e1eb2f9",
			      "pt": "8466e3b1680b40f8bae488e2f392ab33",
			      "ct": "161ac86ab184f638d43b1435161ebec4"
                            },
                            {
			      "count" : 1,
			      "key": "9090af6f71ba953bf31e88e5887d8b8e",
			      "iv": "161ac86ab184f638d43b1435161ebec4",
			      "pt": "b9cc35ba5d5ea63cf6516a200a17c36b",
			      "ct": "8c2998edf29357389483bcf33a6484df"
                            },
			  ]
		       }	
                    ]
		  }
                ]
```

**Comment by bfussell** [comment]
Here is a CBC MCT  example from a CAVS response file:

COUNT = 0
KEY = 868a6705c03e630327259cd09e63354a
IV = a771c0fc42300fa7b4c1eca82e1eb2f9
PLAINTEXT = 8466e3b1680b40f8bae488e2f392ab33
CIPHERTEXT = 161ac86ab184f638d43b1435161ebec4

IV does exist for CBC, maybe you are thinking of ECB.  We should point out the difference.
Having the lengths can be helpful for cross checking the data so JSON corruption can
be caught early.

**Comment by Kritner** [comment]
The `iv` itself is indeed needed as a part of the algo input for cbc. The `ivLen` however, is not needed as a parameter for registration, or as a part of the group information, as it is restricted to the block size of 128.  

A similar argument can be made for the `ptLen` within the test group for the top example.  In many cases the `ptLen` for tests is either:

1. At the block size of 128 bytes
2. Empty
2. Variable - MMT tests where the PT len doubles after each test case (IIRC)

**Comment by atvassilev** [comment]
Yes, ptLen varies according to the algorithm. The client can set it however they choose but the server shall validate that the supplied length, if needed, is correct. and return an error if it is not If ptLen is not needed the server shall ignore the value.

**Comment by bfussell** [comment]
So it appears we have a couple items to clean up.  We don't need to include ivlen in the capabilities
or ivlen in the vectors sent by the server except for AEAD ciphers. Non-AEAD ciphers on the client will make the assumption that ivlen = keylen.  

Does that cover the concern ?

As a part of this I plan to re-organize the appendix sections to group the GCM and CBC MCT exchanges separately to make it more clear.

**Comment by atvassilev** [comment]
Resolved with PR #138 .
